File: Chatify(WhatsApp Clone)/app/src/main/java/com/example/whatsapp/activity/MainActivity.java
package com.example.whatsapp.activity;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.provider.MediaStore;
import android.util.Log;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.example.whatsapp.R;
import com.example.whatsapp.adapters.FragmentAdapter;
import com.example.whatsapp.chat_gpt.activity.ChatWithExpertActivity;
import com.example.whatsapp.chat_gpt.activity.GenerateImageActivity;
import com.example.whatsapp.databinding.ActivityLoginBinding;
import com.example.whatsapp.databinding.ActivityMainBinding;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.database.FirebaseDatabase;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding mainBinding;
    FirebaseAuth firebaseAuth;
    private static final int CAMERA_REQUEST = 11;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mainBinding= ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(mainBinding.getRoot());

        getSupportActionBar().setElevation(0);

        firebaseAuth=FirebaseAuth.getInstance();
        mainBinding.viewPager.setAdapter(new FragmentAdapter(getSupportFragmentManager()));
        mainBinding.tabLayout.setupWithViewPager(mainBinding.viewPager);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater=getMenuInflater();
        inflater.inflate(R.menu.menu,menu);
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
         switch (item.getItemId())
         {

             case R.id.camera:
                 openCamera();
                 break;

             case R.id.search:
                 Toast.makeText(this, "Searching...", Toast.LENGTH_SHORT).show();
                 break;

             case R.id.group:
                 Intent intent=new Intent(MainActivity.this,GroupChatActivity.class);
                 startActivity(intent);
                 break;

             case R.id.bot:
                 startActivity(new Intent(getApplicationContext(),ChatWithExpertActivity.class));
                 break;

             case R.id.chatGpt:
                 startActivity(new Intent(getApplicationContext(), GenerateImageActivity.class));
                 break;

             case R.id.setting:
                 startActivity(new Intent(getApplicationContext(),SettingActivity.class));
                 break;

             case R.id.back:
                 mainBinding.mainLayout.setBackgroundResource(R.color.dark_back);
                 break;

             case R.id.normal:
                 mainBinding.mainLayout.setBackgroundResource(R.color.chat_background);
                 break;

             case R.id.white:
                 mainBinding.mainLayout.setBackgroundResource(R.color.white);
                 break;

             case R.id.logout:
                 logOut();

                 break;


         }
        return super.onOptionsItemSelected(item);
    }

   private void logOut()
    {
        new AlertDialog.Builder(this)
                .setTitle("Logout")
                .setMessage("Are you sure want to be logout?")
                .setPositiveButton("Yes", new DialogInterface.OnClickListener() {

                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        firebaseAuth.signOut();
                        startActivity(new Intent(getApplicationContext(),LoginActivity.class));
                    }


                }).setNegativeButton("No", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        dialog.dismiss();
                    }
                }).show();
    }

    private void openCamera()
    {
        Intent intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
        startActivityIfNeeded(intent, CAMERA_REQUEST);
    }

}

File: GalaxyStore/galaxystoreadmin/src/main/java/com/example/galaxystoreadmin/activity/MainActivity.kt
package com.example.galaxystoreadmin.activity

import android.annotation.SuppressLint
import android.app.AlertDialog
import android.content.Intent
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.widget.Toast
import androidx.activity.viewModels
import androidx.drawerlayout.widget.DrawerLayout
import androidx.fragment.app.viewModels
import androidx.navigation.findNavController
import com.google.android.material.navigation.NavigationView
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.navigateUp
import androidx.navigation.ui.setupActionBarWithNavController
import androidx.navigation.ui.setupWithNavController
import com.example.galaxystoreadmin.R
import com.example.galaxystoreadmin.databinding.ActivityMainBinding
import com.example.galaxystoreadmin.utils.Utils
import com.example.galaxystoreadmin.viewmodels.ProductViewModel
import com.google.firebase.auth.FirebaseAuth


class MainActivity : AppCompatActivity() {

    private lateinit var appBarConfiguration: AppBarConfiguration
    private lateinit var binding: ActivityMainBinding
    private val productViewModel: ProductViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)


        setSupportActionBar(binding.appBarMain.toolbar)
        supportActionBar!!.setBackgroundDrawable(ColorDrawable(Color.parseColor("#FFA500")))

        val drawerLayout: DrawerLayout = binding.drawerLayout
        val navView: NavigationView = binding.navView
        val navController = findNavController(R.id.nav_host_fragment_content_main)


        appBarConfiguration = AppBarConfiguration(setOf(
                R.id.home, R.id.addProduct, R.id.order), drawerLayout)

        setupActionBarWithNavController(navController, appBarConfiguration)
        navView.setupWithNavController(navController)

    }

    override fun onSupportNavigateUp(): Boolean {
        val navController = findNavController(R.id.nav_host_fragment_content_main)
        return navController.navigateUp(appBarConfiguration) || super.onSupportNavigateUp()
    }


    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.main_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {

        when(item.itemId)
        {
            R.id.action_logout-> {  logout() }
            R.id.action_profile -> { userAccount()}
        }

        return super.onOptionsItemSelected(item)


    }

    @SuppressLint("SuspiciousIndentation")
    private fun logout() {
        val builder: AlertDialog.Builder = AlertDialog.Builder(this)
        val alertDialog: AlertDialog = builder.create()

           builder.setIcon(R.drawable.logout_icon)
            .setTitle("Logout")
            .setMessage("Are you sure want to be logout?")

            .setPositiveButton("Yes") { _, _ ->
            FirebaseAuth.getInstance().signOut()
            Utils.showToast(this,"You have been logout successfully")
            startActivity(Intent(this,AdminAuthActivity::class.java))
            finish() }

        .setNegativeButton("No"){_,_ ->
          alertDialog.dismiss()
        }
        .show()
        .setCancelable(false)
    }

    private fun userAccount(){
       productViewModel.getAdminAccount {
           Utils.showToast(this,it!!) }
    }
}